src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/MethodBodyAnalyzer.cs

```diff
@@ -429,7 +429,7 @@ private MethodAnalysisContext(IParameterSymbol asyncParameter, Scope scope)
             }
 
             public MethodAnalysisContext WithNewMethod(IMethodSymbol method, IParameterSymbol asyncParameter) 
-                => new MethodAnalysisContext(AsyncParameter ?? asyncParameter, GetScope(method, asyncParameter));
+                => new MethodAnalysisContext(asyncParameter ?? AsyncParameter, GetScope(method, asyncParameter));
```

Comment:
`async` parameter that is defined in local function should hide parameter from parent method.